membership-common/src/main/scala/com/gu/memsub/subsv2/services/CatalogService.scala

```diff
@@ -42,125 +34,23 @@ object FetchCatalog {
 
 }
 
-class CatalogService[M[_]: Monad](productIds: ProductIds, fetchCatalog: M[String \/ JsValue], unsafeGet: M[Catalog] => Catalog, stage: String) {
+object CatalogService extends SafeLogging {
 
-  case class ErrorReport(problem: String, underlying: Map[ProductRatePlanId, String]) {
-    override def toString = s"\n$problem\n\n--- HERE IS WHAT WE TRIED --\n" +
-      underlying.foldLeft("") { case (a, (id, str)) => s"$a${id.get}: $str\n" }
-  }
+  def read[M[_]: Functor](fetchCatalog: M[String \/ JsValue]): M[CatalogMap] = {
 
-  /*
-  frontend id is to prioritise searching for tagged rateplans so we can look for 6 for 6 or quarterly and find the tagged one, but since most of the others won't
-  be tagged, we need to fall back for untagged ones.
-  in time, I expect them all to be requested by the tag, so we can make frontend id non optional
-  in time, I also expect they will all be tagged in zuora, so we don't need to then try untagged ones
-   */
-  def one[P <: ProductRatePlan[Product, RatePlanChargeList, Status]](plans: List[CatalogZuoraPlan], name: String, frontendId: FrontendId)(implicit
-      p: CatPlanReads[P],
-  ): Validation[NonEmptyList[String], P] = {
-    many(plans.filter(_.frontendId.contains(frontendId)), name).flatMap {
-      case a if a.size == 1 => Validation.s(a.head)
-      case e => Validation.failureNel(s"Too many plans! for name $name with frontendId $frontendId $e")
+    val catalog: M[String \/ CatalogMap] = {
+      fetchCatalog.map(_ => ())
+      for {
+        failableCatalog <- fetchCatalog
+      } yield for {
+        catalog <- failableCatalog
+        plans <- Json.fromJson[List[CatalogZuoraPlan]](catalog).asEither.toDisjunction.leftMap(_.toString)
+      } yield plans.map(catalogZuoraPlan => catalogZuoraPlan.id -> catalogZuoraPlan).toMap
     }
-  }
-
-  def many[P <: ProductRatePlan[Product, RatePlanChargeList, Status]](plans: List[CatalogZuoraPlan], name: String)(implicit
-      p: CatPlanReads[P],
-  ): ValidationNel[String, NonEmptyList[P]] = {
-    val parsed: List[(ProductRatePlanId, ValidationNel[String, P])] =
-      plans.map(plan => (plan.id, p.read(productIds, plan)))
 
-    val failures = parsed
-      .filter { case (_, validation) => validation.isFailure }
-      .map { case (id, err) => (id, err.swap.map(_.list.toList).getOrElse(List.empty).mkString(", ")) }
-      .toMap
+    catalog.map(_.leftMap[CatalogMap](error => {
+      logger.errorNoPrefix(scrub"error: $error"); Map.empty
```

Comment:
not sure why it returns an empty catalog if there's an error (a Left)?? but that's what it did before.  Ideally it should fail the health check to avoid broken instances coming in service